src/main/java/com/github/manolo8/darkbot/utils/DiscordWebhook.java

```diff
@@ -0,0 +1,238 @@
+package com.github.manolo8.darkbot.utils;
+
+import com.github.manolo8.darkbot.utils.http.Http;
+import com.github.manolo8.darkbot.utils.http.Method;
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+import java.awt.*;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Adaptation of original source: https://gist.github.com/k3kdude/fba6f6b37594eae3d6f9475330733bdb
+ */
+public class DiscordWebhook {
+    private static final Gson GSON = new Gson();
+    private final List<EmbedObject> embeds = new ArrayList<>();
+    private String content;
+    private String username;
+    @SerializedName("avatar_url") private String avatarUrl;
+    private boolean tts;
+
+    public void setContent(String content) {
+        this.content = content;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setAvatarUrl(String avatarUrl) {
+        this.avatarUrl = avatarUrl;
+    }
+
+    public void setTts(boolean tts) {
+        this.tts = tts;
+    }
+
+    public void addEmbed(EmbedObject embed) {
+        this.embeds.add(embed);
+    }
+
+    public int execute(String url) throws IOException {
+        if (this.content == null && this.embeds.isEmpty()) {
+            throw new IllegalArgumentException("Set content or add at least one EmbedObject");
+        }
+
+        return Http.create(url, Method.POST)
+                .setRawHeader("Content-Type", "application/json; charset=UTF-8")
+                .setBody(GSON.toJson(this).getBytes(StandardCharsets.UTF_8))
+                .getConnection().getResponseCode();
+    }
+
+    public static class EmbedObject {
+        private final List<Field> fields = new ArrayList<>();
+        private String title;
+        private String description;
+        private String url;
+        private Color color;
+        private Footer footer;
+        private Thumbnail thumbnail;
+        private Image image;
+        private Author author;
```

Comment:
```suggestion

        private String title;
        private String description;
        private String url;
        private Color color;
        private Footer footer;
        private Image image;
        private Thumbnail thumbnail;
        private Author author;
        private final List<Field> fields = new ArrayList<>();
```

Same thing here, with https://discord.com/developers/docs/resources/channel#embed-object

it also seems like some fields have getters and setters scattered in here, make them properly ordered like: getA, setA, getB, setB, currently i can see getAuthor, getFields, setFooter, setAuthor, in that order, which is odd